tck/src/main/java/org/eclipse/microprofile/lra/tck/TckCancelOnTests.java

```diff
@@ -169,12 +171,12 @@ public void notCancelOnFamily5xx() {
     }
 
     /**
-     * See {@link LraCancelOnController#cancelFromRemoteCall()}
+     * See {@link LraCancelOnRemoteCallController#cancelFromRemoteCall()}
      */
     @Test
     public void cancelFromRemoteCall() {
-        Response response = getSuiteTarget().path(LRA_CANCEL_ON_CONTROLLER_PATH)
-                .path(LraCancelOnController.CANCEL_FROM_REMOTE_CALL).request().get();
+        Response response = getSuiteTarget().path(LRA_CANCEL_ON_REMOTE_CALL_CONTROLLER_PATH)
+                .path(LraCancelOnRemoteCallController.CANCEL_FROM_REMOTE_CALL).request().get();
         assertEquals("The 200 status response is expected", Status.OK.getStatusCode(), response.getStatus());
         assertEquals("Status was 200 but compensate should be called twice as LRA should be cancelled for remotely called participant as well",
                 beforeCompensatedCount + 2, getCompensateCount());
```

Comment:
Why do we assume here the compensate is called twice? A single LRA should only be compensated once (in the case the first call returned status 200 as we do here)

Reply:
there are 2 participants subscribed within this test:

test calls A, A calls B, B returns cancel on status (5xx) so the LRA should be cancelled. So both A and B should call compensate

Reply:
What is the status of this PR. To me the test looks wrong, the same resource is invoked twice in the context of the same LRA and therefore there should be only a single participant registered with the LRA, hence a single compensate callback. I recommend either deleting the test or replace it with something that we aren't already testing.

Reply:
An analogy with XA would be if a client started a transaction and made two updates to a database and to expect that the XA resource is asked to commit twice for the same transaction which would be incorrect behaviour.

Reply:
My advice is to change the test to assert that the compensate call is only issued once:

```java
assertEquals("Compensate was called more than once on the same registered participant",
                beforeCompensatedCount + 1, getCompensateCount());
```
Then the test is a good test because it validates that the implementation does not issue duplicate compensate notifications (I don't believe we currently have a test for that).

Reply:
@mmusgrov your assumption is not right, there are two resources enlisted within the LRA -- LraCancelOnRemoteCallController and LraCancelOnController. They just both share an application scoped bean ParticipantService which serves as a shared counter to verify that compensate has been called on both participants. 

I am not saying the test case is right. It is something that was already accepted as part of cancel on tests. This is really just refactoring because as we've discussed on the meeting the same participant cannot be enlisted within the same LRA twice.

Reply:
I am looking at the code in the cancelFromRemoteCall test that is currently in the repo which only calls a single participant. If you have changed the test to call two different participants then I would have thought we already have a test for that. We should just change the assert along the lines I suggested above.

Reply:
@xstefank I missed that  ParticipantService is shared between the two resources.

But the test is not deterministic then. Was there a compensate for A and B? or was it just Compensate B which was called twice, or Compensate A twice?

Reply:
@rdebusscher yes you are right. I didn't think that would be a problem. But I can update ParticipantService to take into account where the compensate has been called.

Reply:
@ochaloup Can you raise a bug fix for the existing test to change the assert to 1. That way our implementations will start passing again, That way we can continue this discussion at our leisure.

Reply:
@mmusgrov changing the assert to 1 is not the correct solution. It could be that the Narayana implementation will start to pass but the test will be wrong. We can't drive the specification based on what the implementation expects.

Summary of the issue
* I wrote the test `cancelFromRemoteCall` which tested two separate things "cancel on remote call" and "multiple registration of the same participant"
* this PR changes the test `cancelFromRemoteCall` in the correct way that the test will be testing only the behaviour of the "cancel on remote call"
* the issue #109 was created based on our discussion to clear the specification on this and to fix the failing tests. 
* this  PR was gradually changed to contain only  fix for the test `cancelFromRemoteCall
* this PR fixes the `cancelFromRemoteCall` and I think it fixes it in the good manner
* we possibly should consider adding a specific test for the "clarify multiple registrations by the same participant"
* I would be glad if we can shortly talk on the hangout if the spec is really clear in the manner of the issue #109